### PR TITLE
nodeipam: actually register the nodeipam cell into the operator 

### PR DESCRIPTION
**Pull request by MrFreezeex**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Actually register the nodeipam cell as it was missed in the original PR #30038 
Also add some docs to enable the feature at the helm level since it is disabled by default.

```release-note
Fix nodeipam cell not registered
```

### CODE REVIEW

File: Documentation/network/node-ipam.rst [hunk 1]
```diff
@@ -23,3 +23,22 @@ shall be used and will use the ``ExternalIP`` addresses if any or the
 To use this feature your service must be of type ``LoadBalancer`` and have the
 `loadBalancerClass <https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class>`__
 set to ``io.cilium/node``.
+
+Cilium's node IPAM is disabled by default.
+To install Cilium with the node ipam, run
```
Comment (qmonnet) [inline Documentation/network/node-ipam.rst]:
```suggestion
To install Cilium with the node IPAM, run:
```

File: Documentation/network/node-ipam.rst [hunk 2]
```diff
@@ -23,3 +23,22 @@ shall be used and will use the ``ExternalIP`` addresses if any or the
...
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| \\
+     --namespace kube-system \\
+     --set nodeIPAM.enabled=true
+
+To enable the node ipam on an existing installation, run
```
Comment (qmonnet) [inline Documentation/network/node-ipam.rst]:
```suggestion
To enable the node IPAM on an existing installation, run:
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Other than small minor nits (see below), the PR looks good, thanks a lot!

**Comment by qmonnet** [comment]
This probably calls for backports? Fabio, what do you think?

**Comment by MrFreezeex** [comment]
> This probably calls for backports? Fabio, what do you think?

FYI It's a new feature merged last week (https://github.com/cilium/cilium/pull/30038), so a backport of this particular PR shouldn't be necessary but if you want the feature the "whole feature" to be backported I'm happy to do it as well.

**Review by pippolo84 — APPROVED** [review]
LGTM :100:

**Comment by pippolo84** [comment]
> This probably calls for backports? Fabio, what do you think?

I think we don't need it, https://github.com/cilium/cilium/pull/30038 is not part of v1.15 or any other stable branch.

**Review by qmonnet — APPROVED** [review]
Thanks!

**Pull request merged**